pkg/roachprod/roachprod.go

```diff
@@ -1642,6 +1642,9 @@ func Grow(ctx context.Context, l *logger.Logger, clusterName string, numNodes in
 	if err != nil {
 		return err
 	}
+	if c.IsLocal() {
+		return LoadClusters()
```

Comment:
This logic could some use explanation / comments.

Reply:
Done. This is actually only needed if you call this from a `roachtest`, or intended to do anything with the cluster directly after, since the cache needs updating.